### ISSUE TITLE
RewardedVideoAd support 

### ISSUE THREAD

**Issue by dr4mohamed** [MichaelStott/KivMob#9]
hi,
please can you add RewardedVideoAd support to kivmob !

**Comment by MichaelStott** [comment]
Initial support for rewarded video ads has been added in KivMob 2.0. Still ironing out a few details, but they can be displayed. Will update here shortly.

**Comment by MichaelStott** [comment]
Callbacks for rewarded ads are now fully supported via users passing in an object implementing RewardedListenerInterface to _set_rewarded_ad_listener_ function.

**Comment by Denis-Memo** [comment]
Hallow. Can you please give an example on how to use `RewardedListenerInterface` because this line

```
# Add any callback functionality to this class.
self.ads.set_rewarded_ad_listener(RewardedListenerInterface())

``` 
is almost useless to me without a working example on how to get Rewarded Callbacks. You never give an example on how to use that.

**Comment by MichaelStott** [comment]
Sorry to hear that, could elaborate on what you need clarified? Basically, you would create an object implementing that interface and then pass that into the set_rewarded_ad_listener function.

**Comment by Denis-Memo** [comment]
MichaelScott, i am not a very good Python programmer i code mostly in java but i love Python so much, and that is why i want to use Kivy on android instead of java. What am having trouble with is getting Kivmob callback to my main.py. Lets say i want to know if a person watch an ad to the end or if an ad is ready. How can i get KivMob to tell me that?

Something like this will be really usefully to me.

```
self.a = self.ads.set_rewarded_ad_listener(RewardedListenerInterface())
if self.a.on_rewarded(): #I know this is dumb but i wish to learn
    self.giveReward()
```

**Comment by akgupta0777** [comment]
> Sorry to hear that, could elaborate on what you need clarified? Basically, you would create an object implementing that interface and then pass that into the set_rewarded_ad_listener function.

@MichaelStott  
I implemented what you said in this quoted Reply 
Here what i done so far by my understanding...
I am using your Demo App Code for testing Ads and all stuff.
**Here I am trying to implement RewardedAd Listener**

`  
   **class KivMobDemo(MDApp):**
    
    def __init__(self,**kwargs):
        self.theme_cls.theme_style = "Dark"
        super().__init__(**kwargs)        

    # These are our Admob Ad IDs
    APP = "ca-app-pub-XXXXXXXXXX"
    BANNER = "ca-app-pub-XXXXXXXXXXXX"
    INTERSTITIAL = "ca-app-pub-XXXXXXXXX"
    REWARDED_VIDEO = "ca-app-pub-XXXXXXXX"
    TEST_DEVICE_ID = "3C91BAC5C088815B62389497AC1E309D"
    Count = 0
    show_banner = False
    ads = KivMob(APP)
    def build(self):
        self.ads.add_test_device(self.TEST_DEVICE_ID)
        self.ads.new_banner(self.BANNER, False)
        self.ads.new_interstitial(self.INTERSTITIAL)
        self.ads.request_banner()
        self.ads.request_interstitial()
        self.ads.set_rewarded_ad_listener(RewardsHandler()) #See This Line, Do i did Right
        self.ads.load_rewarded_ad(self.REWARDED_VIDEO)
        self.toggled = False
        self.ads.show_banner()
        return KivMobDemoUI()

    def toggle_banner(self):
        self.show_banner = not self.show_banner
        if self.show_banner:
            self.ads.show_banner()
         else:
             self.ads.hide_banner()

    def load_video(self):
        self.ads.load_rewarded_ad(self.REWARDED_VIDEO)

**Made A Seperate Class For Overriding Callback Functions**

**class RewardsHandler(RewardedListenerInterface):**

    def on_rewarded(self, reward_name, reward_amount):
        KivMobDemo.Count += reward_amount
        print("User rewarded")

    def on_rewarded_video_ad_completed(self):
        self.on_rewarded("Reward",1)
        print("Ad Completed Time to give Rewards")

    def on_rewarded_video_ad_closed(self):
        KivMobDemo.Count+=0

    def on_rewarded_video_ad_left_application(self):
        KivMobDemo.Count = 0

    def on_rewarded_video_ad_started(self):
        KivMobDemo.load_video()

    def on_rewarded_video_ad_failed_to_load(self, error_code):
        if(error_code == 0):
            print("Something Went Wrong, Please Try Again")
        elif(error_code == 1):
            print("Please Report To the Developer")
        elif(error_code == 2):
            print("Make Sure You Have Internet Access")
        else:
            print("Please Try Later")
`
**Kivy Code** 
**Basically i am trying to increase the count variable after the rewarded video played.** 
```
            Screen:
                name: 'rewarded'
                BoxLayout:
                    MDLabel:
                        font_style: 'H1'
                        theme_text_color: 'Primary'
                        text: "Counter "+str(app.Count)
                        halign: 'center'
                        pos_hint: {'center_x': 0.5, 'center_y': 0.75}
                MDFloatingActionButton:
                    icon: 'plus'
                    elevation_normal: 2
                    pos_hint: {'center_x': 0.5, 'center_y': 0.25}
                    on_release: 
                        app.ads.show_rewarded_ad()
```

**Comment by Afani-Ctrl** [comment]
Would the Count variable be similar to my score (keeping track of the score once player meets scoring requirements)

**Comment by akgupta0777** [comment]
@Afani-Ctrl yeah you can do anything with rewarded video ads functions. The above example is just a basic implementation of how to add rewarded video ads in your app.

For more info 
Watch : https://youtube.com/playlist?list=PLbjTljak8oh4VBNfZTv9QBOsljaWOPE-p
